packages/help-center/src/hooks/use-messaging-auth.ts

```diff
@@ -0,0 +1,40 @@
+import config from '@automattic/calypso-config';
+import { useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+import type { MessagingAuth } from '../types';
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
+async function requestMessagingAuth() {
+	const currentEnvironment = config( 'env_id' );
+	const params = { type: 'zendesk', test_mode: String( currentEnvironment === 'development' ) };
+	const wpcomParams = new URLSearchParams( params );
+	return canAccessWpcomApis()
+		? ( ( await wpcomRequest( {
+				path: '/help/authenticate/chat',
+				query: wpcomParams.toString(),
+				apiNamespace: 'wpcom/v2',
+				apiVersion: '2',
+				method: 'POST',
+		  } ) ) as MessagingAuth )
+		: ( ( await apiFetch( {
+				path: addQueryArgs( '/help-center/authenticate/chat', params ),
+				method: 'POST',
+				global: true,
+		  } as APIFetchOptions ) ) as MessagingAuth );
+}
```

Comment:
The `async` is creating a promise around another promise. 

```suggestion
function requestMessagingAuth() {
	const currentEnvironment = config( 'env_id' );
	const params = { type: 'zendesk', test_mode: String( currentEnvironment === 'development' ) };
	const wpcomParams = new URLSearchParams( params );
	return canAccessWpcomApis()
		? wpcomRequest<MessagingAuth>( {
				path: '/help/authenticate/chat',
				query: wpcomParams.toString(),
				apiNamespace: 'wpcom/v2',
				apiVersion: '2',
				method: 'POST',
		  } ) 
		: apiFetch<MessagingAuth>( {
				path: addQueryArgs( '/help-center/authenticate/chat', params ),
				method: 'POST',
				global: true,
		  } as APIFetchOptions ) ;
}
```

Reply:
Thanks for pointing that (and similar instances) out! I was copying code around and looks like it sneaked in. Fixed in https://github.com/Automattic/wp-calypso/pull/76777/commits/bcbcc946e4440b6c606700fea168825b586e6007